src/Lib/Components/Size.vala

```diff
@@ -41,7 +41,12 @@ public class Akira.Lib.Components.Size : Component {
 
             if (locked && !auto_resize) {
                 auto_resize = true;
-                height = Utils.AffineTransform.fix_size (value / ratio);
+                if (item is Items.CanvasImage && Utils.AffineTransform.fix_size (value * ratio) < 1) {
+                    height = 1;
+                }
+                else {
+                    height = Utils.AffineTransform.fix_size (value * ratio);
+                }
```

Comment:
Good solution but we can clean up the code a bit to avoid repetition and unnecessary double calculations.
```suggestion
var locked_height = Utils.AffineTransform.fix_size (value / ratio);
height = item is Items.CanvasImage && locked_height < 1 ? 1 : locked_height;
```